packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -935,7 +938,10 @@ export class ConnectionManager implements IConnectionManager {
 		// should probably live in the driver.
 		await waitForOnline();
 
-		this.triggerConnect(requestedMode);
+		this.triggerConnect(
+			error !== undefined ? "Reconnect on Error" : disconnectMessage,
```

Comment:
Maybe instead of disconnectMessage we should use something like `Reconnect due to ${disconnectMessage}` or something like that, such that it's more obvious when telemetry where it comes from?